Inventory.cs

```diff
@@ -0,0 +1,126 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace InventoryManagementSystem
+{
+    public class Inventory
+    {
+        private List<Product> Products  = new List<Product>();
+
+        public void AddProduct()
+        {
+            Console.Write("Enter the name of the product: ");
+            string? name = Console.ReadLine();
+
+            Console.Write("Enter the price of the product: ");
+            double price;
+            while (!double.TryParse(Console.ReadLine(), out price))
+            {
+                Console.WriteLine("Invalid input! Please enter a valid price.");
+            }
+
+            Console.Write("Enter the quantity of the product: ");
+            int quantity;
+            while (!int.TryParse(Console.ReadLine(), out quantity))
+            {
+                Console.WriteLine("Invalid input! Please enter a valid quantity.");
+            }
+
+            Products.Add(new Product(name, price, quantity));
+            Console.WriteLine($"Whooooh! A new Product added with name: {name}, price: {price}, and quantity: {quantity}");
+        }
+        public void ViewAllProducts()
+        {
+            int listSize = Products.Count;
+            Console.WriteLine($"\n We have {listSize} products in our inventory\n");
+            Console.WriteLine("--------------------------------------------------------------------");
+            Console.WriteLine("|       Name       |       Price       |       Quantity       |");
+            Console.WriteLine("--------------------------------------------------------------------");
+            foreach (Product product in Products )
+            {
+                Console.WriteLine($"| {product.Name,-15} | {product.Price,-17} | {product.QuantityInStock,-20} |");
+            }
+            Console.WriteLine("--------------------------------------------------------------------");
+        }
+
+        public void EditProduct()
+        {
+            Console.Write("Enter the name of the product to edit: ");
+            string? name = Console.ReadLine();
+
+            foreach (Product product in Products)
+            {
+                if (product.Name.Equals(name))
```

Comment:
Note, equals comparing 2 strings with case sensitive so you should pass the a comaprer type like this if you want to compare with incase sensitive
```suggestion
                if (product.Name.Equals(name, , StringComparison.OrdinalIgnoreCase))
```